proposals/user-scripts-api.md

```diff
@@ -157,11 +162,54 @@ As mentioned in requirement A, the user script world can communicate with differ
   - Scripts registered via [`scripting.registerContentScripts()`](https://developer.chrome.com/docs/extensions/reference/scripting/#method-registerContentScripts), following the order they were registered in. Updating a content script doesn't change its registration order.
   - Scripts registered via `userScripts.register()`, following the order they were registered in. Updating a user script doesn’t change its registration order.
 - User scripts are always persisted across sessions, since the opposite behavior would be uncommon. (We may explore providing an option to customize this in the future.)
+- Unlike regular content scripts, `matches` is allowed to be optional when `includeGlobs` is specified. A user script matches a document when its URL matches either `matches` or `includeGlobs`.
+
+### RegisteredUserScript validation
+
+The `RegisteredUserScript` type is shared by `userScripts.register()` and
+`userScripts.update()`. All fields except `id` are declared as optional, to
+allow `userScripts.update()` to update individual properties.
+
+The following fields are required for a valid `RegisteredUserScript`:
+
+- `js` must be a non-empty array.
+- At least one of `matches` or `includeGlobs` must be a non-empty array.
+
+The above requirements must be checked when `userScripts.register()` is called.
+
+When `userScripts.update()` is invoked, individual properties may be `null` or
```

Comment:
Chrome currently resets properties when individual properties are missing, but there is consensus that it should not be changed. For details, see https://github.com/w3c/webextensions/issues/565#issuecomment-2536646436

... and this is tracked at https://issues.chromium.org/issues/40938749